frontend/src/Editor/Editor.jsx

```diff
@@ -1752,41 +1669,36 @@ class EditorComponent extends React.Component {
-                      <Inspector
-                        moveComponents={this.moveComponents}
-                        componentDefinitionChanged={this.componentDefinitionChanged}
-                        removeComponent={this.removeComponent}
-                        selectedComponentId={selectedComponents[0].id}
-                        allComponents={appDefinition.pages[this.state.currentPageId]?.components}
-                        key={selectedComponents[0].id}
-                        switchSidebarTab={this.switchSidebarTab}
-                        apps={apps}
-                        darkMode={this.props.darkMode}
-                        appDefinitionLocalVersion={this.state.appDefinitionLocalVersion}
-                        pages={this.getPagesWithIds()}
-                        cloneComponents={this.cloneComponents}
-                      ></Inspector>
-                    ) : (
-                      <center className="mt-5 p-2">
-                        {this.props.t('editor.inspectComponent', 'Please select a component to inspect')}
-                      </center>
-                    )}
-                  </div>
-                )}
-
-                {currentSidebarTab === 2 && (
-                  <WidgetManager
-                    componentTypes={componentTypes}
-                    zoomLevel={zoomLevel}
-                    darkMode={this.props.darkMode}
-                  ></WidgetManager>
-                )}
+                <RightSidebarTabManager
+                  inspectorTab={
+                    <div className="pages-container">
+                      {!isEmpty(appDefinition.pages[this.state.currentPageId]?.components) ? (
+                        <Inspector
+                          moveComponents={this.moveComponents}
+                          componentDefinitionChanged={this.componentDefinitionChanged}
+                          removeComponent={this.removeComponent}
+                          allComponents={appDefinition.pages[this.state.currentPageId]?.components}
+                          apps={apps}
+                          darkMode={this.props.darkMode}
+                          appDefinitionLocalVersion={this.state.appDefinitionLocalVersion}
+                          pages={this.getPagesWithIds()}
+                          cloneComponents={this.cloneComponents}
+                        />
+                      ) : (
+                        <center className="mt-5 p-2">
+                          {this.props.t('editor.inspectComponent', 'Please select a component to inspect')}
+                        </center>
+                      )}
+                    </div>
+                  }
+                  widgetManagerTab={
+                    <WidgetManager
+                      componentTypes={componentTypes}
+                      zoomLevel={zoomLevel}
+                      darkMode={this.props.darkMode}
+                    />
```

Comment:
This is okay for now. Once app def change is merged, we can separate all the logic out of the components